### PR TITLE
update to swift 4.2

### PR DESCRIPTION
**Pull request by moldovaniosif**
What does this implement/fix? Explain your changes.
---------------------------------------------------
Update project to support swift 4.2.

Does this close any currently open issues?
------------------------------------------
No.

Any relevant logs, error output, etc?
-------------------------------------

Any other comments?
-------------------
…

Where has this been tested?
---------------------------
**Devices/Simulators:** iPhone 7, Simulator 7, X;

**iOS Version:** 12.0

**Swift Version:** 4.2

**MessageKit Version:** 1.0.0

### CODE REVIEW

File: Sources/Views/PlayButtonView.swift [hunk 1]
```diff
@@ -28,7 +28,7 @@ open class PlayButtonView: UIView {
 
     // MARK: - Properties
 
-    open let triangleView = UIView()
+    public let triangleView = UIView()
```
Comment (iceman201) [inline Sources/Views/PlayButtonView.swift:31]: “open” is now simply “more public than public”, providing a very simple and clean model.

**Review by iceman201 — CHANGES_REQUESTED** [review]

**Comment by aaronhuisinga** [comment]
Definitely could used this merged. Unable to compile in Xcode 10 without the update to Swift 4.2.

**Comment by nathantannar4** [comment]
Tests failing

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by brandon-haugen** [comment]
@nathantannar4, it looks like this is only failing because it’s using Xcode 9 (which does not support Swift 4.2) instead of using Xcode 10 during CircleCI builds.

**Comment by nathantannar4** [comment]
Correct, the CircleCi file needs to be updated along with this PR. Also, target branch should be development not master @moldovaniosif. Please make sure your changes are rebased onto development as we already have pending changes 

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by brandon-haugen** [comment]
@nathantannar4 thank you! I wasn’t sure what need to be done to update Circle CI to use the updated version of Xcode. I meant to add that in my first comment since others might not be familiar with Circle CI either.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by SD10** [comment]
Closing in favor of #883. @moldovaniosif appreciate you taking the time to do this 💪 Sent you an invite to the MessageKit org

**Pull request closed without merging**